composer.json

```diff
@@ -82,17 +82,18 @@
         "symfony/expression-language": "^6.1 || ^7.0",
         "symfony/finder": "^6.1 || ^7.0",
         "symfony/form": "^6.1 || ^7.0",
-        "symfony/framework-bundle": "^6.1 || ^7.0",
+        "symfony/framework-bundle": "^6.4.3 || ^7.0",
```

Comment:
let's target main or remove composer changes?

Reply:
Updated to `main`. There is no way to get `--prefer-lowest` to pass without these.

Reply:
but is this tight to enums?

Reply:
`symfony/framework-bundle` change seems to be not needed now the rest of the SF components are `^6.4` so I've dropped it.

`symfony/phpunit-bridge` is to include https://github.com/symfony/symfony/issues/52844 which meant the tests just wouldn't run
```shell
Warning: The lock file is not up to date with the latest changes in composer.json. You may 
be getting outdated dependencies. It is recommended that you run `composer update` or 
`composer update <package name>`.
- Required package "symfony/phpunit-bridge" is not present in the lock file.
```

`symfony/string` is 100% about enums, and was the hardest to track down at the time. The change does mirror the constraint used in the component `composer.json` files.

Reply:
Anticipating the question "Why isn't CI failing?" (I had that question too), see this example from `main` yesterday:
https://github.com/api-platform/core/actions/runs/8679784849/job/23799224716?pr=6309
> PHPUnit is not installed. Please run ./vendor/bin/simple-phpunit to install it

There seems to be a new-ish issue with CI set-up.

Originally I caught the failure on CI and then moved to local testing where it work correctly with `6.4.1`+

Reply:
See #6113